File: vada_x.py
import sys

n = int(sys.stdin.readline()) 

songs = {}
for i in range(n):
    [group, num_str] = sys.stdin.readline().split()
    songs[group] = int(num_str)
    
m = int(sys.stdin.readline())
likes = {}
for i in range(m):
    group = sys.stdin.readline().strip()
    if group in likes:
        likes[group] += 1
    else:
        likes[group] = 1

for group in songs.keys():
    if (songs[group] > likes[group]):
        sys.stdout.write(group)
        break
else:
    sys.stdout.write('NO KPOP FOR VADER')

sys.stdout.write('\n')

File: Vada_x.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashMap;
import java.util.StringTokenizer;

public class Vada_x {
	public static void main(String[] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
        HashMap<String, Integer> groups = new HashMap();
        StringTokenizer st;
		for (int i=0; i<n; i++){
			st = new StringTokenizer(br.readLine());
            String group = st.nextToken();
            int songs = Integer.parseInt(st.nextToken());
            groups.put(group, songs);

		}
        int t = Integer.parseInt(br.readLine()); 
        for (int i=0; i<t; i++){
            String group = br.readLine();
            if (groups.containsKey(group)) {
                groups.put(group, groups.get(group) - 1);
            }
        }
        boolean flag = false;
        for (String group: groups.keySet()){
            if (groups.get(group) > 0) {
                flag = true;
                System.out.println(group);
                break;
            }
        }
        if (!flag) {
            System.out.println("NO KPOP FOR VADER");
        }
	}
}